reference/services/fetch-service.rst

```diff
@@ -0,0 +1,149 @@
+   Launchpad sends those details to 
+   `launchpad buildd <https://git.launchpad.net/~launchpad/launchpad-buildd>`_
+   to be used during the build.
+
+2. During a build, any external requests go through the fetch service proxy
+   endpoint, using the specific ``session ID`` and ``token``. The fetch
+   service keeps track of all requests made during the build to that given
+   ``session ID``.
+
+3. After the pull-phase or at the end of the build (depending on the
+   configuration of the snap itself within Launchpad), `launchpad-buildd`
+   makes a request to the fetch service control endpoint to revoke the
+   ``token``, meaning no new external requests can be made during that build.
+
+4. After the end of the build, when Launchpad gathers the resulting artifacts,
+   Launchpad makes a request to the fetch service control endpoint to gather
+   the metadata from the session (i.e., a file that contains all the requests
+   and dependencies fetched during the build) for the specified ``session ID``.
+
+5. Launchpad makes a final request to the fetch service control endpoint to
+   end session ``session ID``, which deletes all data for that session that
+   was stored in the fetch service database.
+
+Using the fetch service
+~~~~~~~~~~~~~~~~~~~~~~~
+Currently, only snaps can use the fetch service.
+
+To do so, a Launchpad admin is required to set the
+``use_fetch_service`` flag within a snap to ``true``, either in the API or in
+the UI by accessing the Admin area of a snap.
+
+Requests to use the fetch service shall be made to the Launchpad team through
+the usual channels (by
+`opening a question <https://answers.launchpad.net/launchpad>`_, or emailing
+the team at feedback@launchpad.net).
+
+Fetch service maintainers
+~~~~~~~~~~~~~~~~~~~~~~~~~
+The fetch service code itself is not maintained by Launchpad - it is
+maintained by Canonical's Starcraft team. Nonetheless, the service was created
+to be used by Launchpad builders, and that's currently its purpose.
+
+Launchpad maintains its mojo specs and deployment to its environments.
+
+Documentation
+-------------
+* `Snap docs <https://github.com/canonical/fetch-service/tree/main/docs>`_
+
+Git repositories
+----------------
+* `Fetch service Snap <https://github.com/canonical/fetch-service>`_
+* `Fetch service Charm <https://github.com/canonical/fetch-operator>`_
+* `lp-fetch-service Mojo spec <https://git.launchpad.net/~launchpad/launchpad-mojo-specs/+git/private/tree/lp-fetch-service>`_
+
+The following repositories also contain code relevant to how Launchpad uses
+the fetch service:
+
+* `buildd <https://git.launchpad.net/~launchpad/launchpad-buildd>`_
+* `launchpad-buildd-manager <https://git.launchpad.net/~launchpad/launchpad/tree/charm/launchpad-buildd-manager>`_
+
+Bug trackers
+------------
+* `Fetch service snap GitHub issues <https://github.com/canonical/fetch-service/issues>`_
+* `Fetch service charm GitHub issues <https://github.com/canonical/fetch-operator/issues>`_
```

Comment:
Sometimes you capitalize snap and charm, sometimes not. I'd assume the capitalized version is the right one.

Reply:
I made it coherent now, I changed it to non-capitalized - I felt it made more sense